### PR TITLE
Jve/cmake integration part2

### PR DESCRIPTION
**Pull request by jveijck**
Updated structure of makefiles to avoid name classes.
Renamed <component-dir>/datatypes.hpp -><component-dir>/<component>_datatypes.hpp 
- Generate source from component proto files from toplevel
CmakeLists.txt to get correct header file guard and restore MRA-component-proto library

Update generator to handle new situation 
- datatypes.hpp -><component>_datatypes.hpp 
- MRA-component-proto reintroduction

- updated bazel build files and CMakeLists.txt for new situation.

### CODE REVIEW

File: base/abstract_interface.hpp [hunk 1]
```diff
@@ -6,32 +6,30 @@
 namespace MRA
 {
 
-template <typename Ti, typename Tp, typename Ts, typename To, typename Tl>
+template <typename T_InputType, typename T_ParamsType, typename T_StateType, typename T_OutputType, typename T_LocalType>
```
Comment (janfeitsma) [inline base/abstract_interface.hpp:9]: T_...Type is a bit redundant. Why not just InputType. Actually, why did you change this from Ti etc?

File: base/abstract_interface.hpp [hunk 2]
```diff
@@ -6,32 +6,30 @@
...
 class MRAInterface
 {
 public:
-
-    using InputType = Ti;
-    using ParamsType = Tp;
-    using StateType = Ts;
-    using OutputType = To;
-    using LocalType = Tl;
+	using InputType = T_InputType;
```
Comment (janfeitsma) [inline base/abstract_interface.hpp]: Not a big fan of tabs, but ok.

File: CMakeLists.txt [hunk 3]
```diff
@@ -73,6 +74,20 @@ foreach (protobuf_file ${MESSAGE_PROTOS_COMMON})
...
 
+##########################################################
+# component proto files
+##########################################################
+# generate source from component proto files from toplevel to get correct header file guard 
+
+# collect proto files from sub-directories below directory component
+FILE(GLOB_RECURSE MESSAGE_PROTOS_COMPONENTS ABSOLUTE ${CMAKE_CURRENT_SOURCE_DIR}/components *.proto)
+# create single library with all generated sources from component proto files
+add_library(MRA-components-proto)
```
Comment (janfeitsma) [inline CMakeLists.txt:85]: Hm, for the long run, this will become a build hotspot. But ok for now I guess.

File: base/abstract_interface.hpp (hunk 1 above)
Comment (jveijck) [reply base/abstract_interface.hpp:9]: because InputType is defined in the class ca 8 lines below this line.

File: base/abstract_interface.hpp (hunk 2 above)
Comment (jveijck) [reply base/abstract_interface.hpp]: agree

File: CMakeLists.txt (hunk 3 above)
Comment (jveijck) [reply CMakeLists.txt:85]: Indeed, it is not preferred. But will solve the issue of creating correct header files.
It is only used for the generating of code from proto.

File: base/abstract_interface.hpp (hunk 2 above)
Comment (jveijck) [reply base/abstract_interface.hpp:13]: it was tabs, changed into spaces
